### DISCUSSION TITLE
automatically make this space read as ""

### DISCUSSION THREAD

**Discussion by yihaoshumi** [jsvine/pdfplumber#710] [Q&A]
[sample.pdf](https://github.com/jsvine/pdfplumber/files/9355267/sample.pdf)
pdf=pdf.open('sample.pdf')
words=pdf.pages[0].extract_words()

How can I automatically make this space read as "". Because this place is actually occupied.
<img width="530" alt="image" src="https://user-images.githubusercontent.com/32957552/185003500-18c0f72f-96a7-4609-9bb1-1548eec05bf2.png">

[{'text': 'text1text2text3',
  'x0': 90.0,
  'x1': 166.30090000000004,
  'top': 78.38544999999999,
  'doctop': 78.38544999999999,
  'bottom': 88.83544999999992,
  'upright': True,
  'direction': 1},
 {'text': '11',
  'x0': 90.0,
  'x1': 101.6477,
  'top': 93.98545000000001,
  'doctop': 93.98545000000001,
  'bottom': 104.43544999999995,
  'upright': True,
  'direction': 1},
 {'text': '22',
  'x0': 117.359,
  'x1': 129.0067,
  'top': 93.98545000000001,
  'doctop': 93.98545000000001,
  'bottom': 104.43544999999995,
  'upright': True,
  'direction': 1},
 {'text': '33',
  'x0': 144.84,
  'x1': 156.48770000000002,
  'top': 93.98545000000001,
  'doctop': 93.98545000000001,
  'bottom': 104.43544999999995,
  'upright': True,
  'direction': 1},
 {'text': '34',
  'x0': 90.0,
  'x1': 101.6477,
  'top': 109.58545000000004,
  'doctop': 109.58545000000004,
  'bottom': 120.03544999999997,
  'upright': True,
  'direction': 1},
 {'text': '35',
  'x0': 117.359,
  'x1': 129.0067,
  'top': 109.58545000000004,
  'doctop': 109.58545000000004,
  'bottom': 120.03544999999997,
  'upright': True,
  'direction': 1},
 {'text': '37',
  'x0': 90.0,
  'x1': 101.6477,
  'top': 125.18545000000006,
  'doctop': 125.18545000000006,
  'bottom': 135.63545,
  'upright': True,
  'direction': 1},
 {'text': '38',
  'x0': 117.359,
  'x1': 129.0067,
  'top': 125.18545000000006,
  'doctop': 125.18545000000006,
  'bottom': 135.63545,
  'upright': True,
  'direction': 1},
 {'text': '39',
  'x0': 144.84,
  'x1': 156.48770000000002,
  'top': 125.18545000000006,
  'doctop': 125.18545000000006,
  'bottom': 135.63545,
  'upright': True,
  'direction': 1}]

**Comment by jsvine** [comment]
Hi @yihaoshumi, and thanks for your interest in this library. The short answer to your question is: this is not possible with `pdfplumber`, since there is no text there. The sense of that part of the page being "occupied" is more an effect of human pattern recognition, and is not encoded in the PDF itself. Depending on the consistency and format of the PDFs you're parsing, however, you may be able to use the positions of the words returned by `.extract_words(...)` to infer where the blank spaces are.

**Reply by yihaoshumi** [reply]
Thank you for your reply. The inference you mentioned, how to deal with this better. Because extract_words is not displayed. Compare the upper and lower distance difference of each character of text2 with the upper and lower distance difference of text3?

**Reply by jsvine** [reply]
Unfortunately, I don't fully understand this question. Can you provide code or screenshots to clarify it?
